p2pnetwork/node.py

```diff
@@ -215,7 +230,11 @@ def run(self):
             try:
                 self.debug_print("Node: Wait for incoming connection")
                 connection, client_address = self.sock.accept()
-                
+
+                if self.check_max_connections():
+                    connection.close()
```

Comment:
Should we accept new connections when the max_connections are reached?  When a connection is closed, this thread is able to start to accept a new connection. Is that possible to do that?

Reply:
You decide the definition of the max connections, but to me make more sense when reach the max connections the node no more accept more inbound and outbound connections. It's prevent a overload in node.

By the logic when disconnect the self.all_nodes list will be one less connections then if someone try to connect will not be blocked.